### PR TITLE
[ZPI-45] - Add new proto operation

### PR DESCRIPTION
**Pull request by Remmo1**
Add new rating operation.

### CODE REVIEW

File: backend/clap/proto/data_store.proto [hunk 1]
```diff
@@ -7,113 +7,129 @@ option java_outer_classname = "DataStoreProto";
...
 
-    repeated AssessorRatings individual = 2;
-    AssessorRatings initial    = 3;
-    AssessorRatings final      = 4;
+  repeated AssessorRatings individual = 2;
+  AssessorRatings initial = 3;
+  AssessorRatings final = 4;
+}
+
+message RatingRequest {
```
Comment (gcfbn) [inline backend/clap/proto/data_store.proto:114]: As we agreed today, the proto file should reflect the following HTTP endpoints:
- POST /submissions/{submissionId}/ratings
with body: {
    ratingType: RatingType;  
}
which will create an empty rating for the given submission. If it's a individual rating, it should retrieve the user ID from the token (as we'll change how we handle authorization). If it's an initial/final rating, it should assign the rating to the leading assessor of the submission.

- POST /partialRatings
with body: {
    criterionId: int?;
    ratingId: int?;
    partialRatingId: int?;
    justification: string;
    points: int;
}
which will perform an upsert of the partial rating. Exactly one of the parameters `ratingId` and `partialRatingId` must be specified. 
If it's `ratingId`, a new partial rating should be created under the specified rating and criterion. 
If it's `partialRatingId`, the partial rating should be updated with the justification and points.

- POST /ratings/{ratingId}/submit which actually doesn't require any body (if you're concerned with POST with empty body, I'm fine to change the interface) and will submit a rating from the draft to the non-draft state. It also includes authorization:
    - only the owner of the individual rating is authorized to submit it
    - only the leading assessor of the initial/final rating is authorized to submit it
and performs a check if all the required partial ratings are present for the rating.

CC: @Veermove pls check if there are no gaps in this contract

PS: the interfaces are something between Java and TS, I hope you will find out what I meant ('?' stands for optional property)

File: backend/clap/proto/data_store.proto [hunk 2]
```diff
@@ -7,113 +7,129 @@ option java_outer_classname = "DataStoreProto";
...
 
 message AssessorRatings {
-    int32 assessor_id = 1;
-    string first_name = 2;
-    string last_name  = 3;
-    repeated PartialRating partial_ratings = 4;
+  int32 assessor_id = 1;
+  string first_name = 2;
+  string last_name = 3;
+  repeated PartialRating partial_ratings = 4;
```
Comment (gcfbn) [inline backend/clap/proto/data_store.proto:103]: Could you please add the `is_draft` here and expose it through the Clap API? @Veermove as it requires your effort too.

**Review by gcfbn — CHANGES_REQUESTED** [review]
Please revert the formatting changes

File: backend/clap/proto/data_store.proto [hunk 3]
```diff
@@ -111,9 +114,23 @@ message RatingsSubmissionResponse {
...
+    int32 points            = 5;
+}
+
+service DataStore {
+    rpc GetSubmissions          (SubmissionRequest)           returns (SubmissionsResponse)       {}
+    rpc GetSubmissionDetails    (DetailsSubmissionRequest)    returns (DetailsSubmissionResponse) {}
+    rpc GetSubmissionRatings    (RatingsSubmissionRequest)    returns (RatingsSubmissionResponse) {}
+    rpc PostNewSubmissionRating (NewSubmissionRatingRequest)  returns (Rating)                    {}
+    rpc PostPartialRating       (PartialRatingRequest)        returns (PartialRating)             {}
+    rpc PostSubmitRating        (google.protobuf.Empty)       returns (PartialRating)             {}
```
Comment (Veermove) [inline backend/clap/proto/data_store.proto:135]: so.. when datastore gets call on this endpoint what is it supposed to do... with no data, no id? When BM said that it requires no body it meant HTTP body. But clap gets `ratingId` from path variable which must me included as grpc request body

**Review by Veermove — APPROVED** [review]

File: backend/clap/proto/data_store.proto [hunk 4]
```diff
@@ -111,9 +112,27 @@ message RatingsSubmissionResponse {
     AssessorRatings final      = 4;
 }
 
-service DataStore {
-    rpc GetSubmissions      (SubmissionRequest)        returns (SubmissionsResponse)       {}
-    rpc GetSubmissionDetails(DetailsSubmissionRequest) returns (DetailsSubmissionResponse) {}
-    rpc GetSubmissionRatings(RatingsSubmissionRequest) returns (RatingsSubmissionResponse) {}
+message NewSubmissionRatingRequest {
```
Comment (gcfbn) [inline backend/clap/proto/data_store.proto:115]: We're missing the submission id. And assessor's id too.

File: backend/clap/proto/data_store.proto [hunk 5]
```diff
@@ -111,9 +112,27 @@ message RatingsSubmissionResponse {
...
+
+message PartialRatingRequest {
+    int32 criterion_id      = 1;
+    int32 rating_id         = 2;
+    int32 partial_rating_id = 3;
+    string justification    = 4;
+    int32 points            = 5;
 }
 
+message UpdateRatingDraft {
```
Comment (gcfbn) [inline backend/clap/proto/data_store.proto:127]: Maybe `SubmitRatingDraft` would be more meaningful?

**Review by gcfbn — CHANGES_REQUESTED** [review]

**Review by gcfbn — APPROVED** [review]

**Pull request merged**